src/history/errors.js

```diff
@@ -1,22 +1,45 @@
-export class NavigationDuplicated extends Error {
-  constructor (normalizedLocation) {
-    super()
-    this.name = this._name = 'NavigationDuplicated'
-    // passing the message to super() doesn't seem to work in the transpiled version
-    this.message = `Navigating to current location ("${
-      normalizedLocation.fullPath
-    }") is not allowed`
-    // add a stack property so services like Sentry can correctly display it
-    Object.defineProperty(this, 'stack', {
-      value: new Error().stack,
-      writable: true,
-      configurable: true
-    })
-    // we could also have used
-    // Error.captureStackTrace(this, this.constructor)
-    // but it only exists on node and chrome
-  }
+function createRouterError (from, to, type, message) {
+  const error = new Error(`${type}: ${message}`)
+  error._isRouter = true
+  error.from = from
+  error.to = to
+  error.type = type
+
+  const newStack = error.stack.split('\n')
+  newStack.splice(1, 2) // remove 2 last useless calls
+  error.stack = newStack.join('\n')
+  return error
+}
+
+export const ErrorTypes = {
+  NAVIGATION_GUARD_REDIRECT: 'NavigationRedirected',
+  NAVIGATION_ABORTED: 'NavigationAborted',
+  NAVIGATION_CANCELLED: 'NavigationCancelled',
+  NAVIGATION_DUPLICATED: 'NavigationDuplicated'
```

Comment:
let's keep the same names as the public API: https://github.com/vuejs/vue-router-next/blob/master/src/errors.ts#L26-L30 (`NavigationFailureType`)